sia/components/hostdb.go

```diff
@@ -20,18 +20,17 @@ const (
 // interfaces.
 
 type HostDB interface {
+	// FlagHost alerts the HostDB that a host is not behaving as expected. The
+	// HostDB may decide to remove the host, or just reduce the weight, or it
+	// may decide to ignore the flagging. If the flagging is ignored, an error
+	// will be returned explaining why.
+	FlagHost(id string) error
```

Comment:
Component functions pretty much always return an error. But perhaps what I've stated in the docstring is a bad approach.